injector/dns_disruption.go

```diff
@@ -134,40 +134,27 @@ func (i *DNSDisruptionInjector) Inject() error {
 	}
 
 	// Set up iptables rules
-	if err := i.config.Iptables.CreateChain(chaostypes.InjectorIptablesChaosDNSChainName); err != nil {
-		return fmt.Errorf("unable to create new iptables chain: %w", err)
-	}
-
-	if err := i.config.Iptables.AddRuleWithIP(chaostypes.InjectorIptablesChaosDNSChainName, "udp", "53", "DNAT", podIP); err != nil {
+	if err := i.config.Iptables.RedirectTo("udp", "53", podIP); err != nil {
 		return fmt.Errorf("unable to create new iptables rule: %w", err)
 	}
 
 	if i.config.Level == chaostypes.DisruptionLevelPod {
 		if !i.config.OnInit {
-			// Redirect traffic marked by targeted InjectorDNSCgroupClassID to CHAOS-DNS
-			if err := i.config.Iptables.AddCgroupFilterRule("nat", "OUTPUT", i.config.Cgroup.RelativePath(""), "-p", "udp", "--dport", "53", "-j", chaostypes.InjectorIptablesChaosDNSChainName); err != nil {
+			// Redirect traffic coming from the targeted container to CHAOS-DNS
+			if err := i.config.Iptables.Intercept("udp", "53", i.config.Cgroup.RelativePath(""), podIP); err != nil {
 				return fmt.Errorf("unable to create new iptables rule: %w", err)
 			}
 		} else {
 			// Redirect all dns related traffic in the pod to CHAOS-DNS
-			if err := i.config.Iptables.AddWideFilterRule("OUTPUT", "udp", "53", chaostypes.InjectorIptablesChaosDNSChainName); err != nil {
+			if err := i.config.Iptables.Intercept("udp", "53", "", ""); err != nil {
 				return fmt.Errorf("unable to create new iptables rule: %w", err)
 			}
 		}
 	}
 
 	if i.config.Level == chaostypes.DisruptionLevelNode {
-		// Exempt chaos pod from iptables re-routing
-		if err := i.config.Iptables.PrependRuleSpec(chaostypes.InjectorIptablesChaosDNSChainName, "-s", podIP, "-j", "RETURN"); err != nil {
-			return fmt.Errorf("unable to create new iptables rule: %w", err)
-		}
-
-		// Re-route all pods under node
-		if err := i.config.Iptables.PrependRuleSpec("OUTPUT", "-p", "udp", "--dport", "53", "-j", chaostypes.InjectorIptablesChaosDNSChainName); err != nil {
-			return fmt.Errorf("unable to create new iptables rule: %w", err)
-		}
-
-		if err := i.config.Iptables.PrependRuleSpec("PREROUTING", "-p", "udp", "--dport", "53", "-j", chaostypes.InjectorIptablesChaosDNSChainName); err != nil {
+		// Re-route all pods under node except for injector pod itself
+		if err := i.config.Iptables.Intercept("udp", "53", "", podIP); err != nil {
```

Comment:
1. We still need it, the pod IP is passed to the `Intercept` function for that reason. In fact, the `Intercept` function embeds the 3 rules (excluding chaos pod IP, prerouting redirect, output redirect).
2. It redirects any DNS packets going through prerouting to the disrupted chain, I don't have the exact details of why we need one in prerouting in addition to the one in output to be honest, I kept the same logic

Reply:
1. Got it, thanks for the explanation.
2. I'm still a bit confused here. By _keeping the same logic_, do you mean the following code in `Intercept` is equivalent to the removed line? 
```
	// inject prerouting rule only if there's no cgroup path filtering
	// packets going through prerouting chain are not yet associated
	// to a process so there's no possibility to filter on a cgroup at this stage
	if cgroupPath == "" {
		if err := i.insert("nat", "PREROUTING", rulespec...); err != nil {
			return err
		}
	}
```

Reply:
Yes. It adds rerouting rules: https://github.com/DataDog/chaos-controller/blob/rework_iptables/network/iptables.go#L129-L141

And it ignores if it's the pod IP: https://github.com/DataDog/chaos-controller/blob/rework_iptables/network/iptables.go#L117-L120

The only difference is that instead of returning for the injector pod IP, it just ignores it in the rule.